gemini/use-cases/document-rag-on-llamaindex/llamaindex_rag.ipynb

```diff
@@ -0,0 +1,1458 @@
+        "    VS_DIMENSIONS = 768\n",
+        "    # Vertex AI Vector Search Index configuration\n",
+        "\n",
+        "    aiplatform.init(project=PROJECT_ID, location=REGION)\n",
+        "\n",
+        "    new_bucket = create_bucket_class_location(GCS_BUCKET)\n",
+        "    vs_index = create_vertex_ai_search_index(VS_INDEX_NAME, VS_DIMENSIONS)\n",
+        "    vs_endpoint = create_vertexai_search_endpoint(VS_INDEX_ENDPOINT_NAME)\n",
+        "    vs_deployed_index = deploy_vertexai_search_endpoint(\n",
+        "        vs_index, vs_endpoint, VS_INDEX_NAME\n",
+        "    )\n",
+        "\n",
+        "    return new_bucket, vs_index, vs_endpoint, vs_deployed_index"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "EUYC_nySmzlK"
+      },
+      "source": [
+        "## Ingest data using LlamaIndex into Vertex AI Vector Search\n",
+        "\n",
+        "The following section leverages LlamaIndex to ingest, chunk, and embed the PDF data to be connected to the Vertex AI Vector Store.\n",
+        "\n",
+        "At the end of this section you will be ready to query against the Vector Store to find relevant context."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 6,
+      "metadata": {
+        "id": "jUK91ivpmzlK"
+      },
+      "outputs": [],
+      "source": [
+        "def initialize_llm_and_storage(vs_index, vs_endpoint):\n",
+        "    \"\"\"\n",
+        "    Initializes Vertex AI Vector Store given a Vertex AI Search index and deployed endpoint.\n",
+        "    Configures embedding and LLMs models to be gecko and Gemini.\n",
+        "    \"\"\"\n",
+        "    # setup storage\n",
+        "    vector_store = VertexAIVectorStore(\n",
+        "        project_id=PROJECT_ID,\n",
+        "        region=REGION,\n",
+        "        index_id=vs_index.resource_name,\n",
+        "        endpoint_id=vs_endpoint.resource_name,\n",
+        "        gcs_bucket_name=GCS_BUCKET,\n",
+        "    )\n",
+        "\n",
+        "    # set storage context\n",
+        "    storage_context = StorageContext.from_defaults(vector_store=vector_store)\n",
+        "\n",
+        "    gemini_embedding_model = VertexTextEmbedding(\"text-embedding-004\")\n",
+        "    llm = Vertex(\"gemini-pro\")\n",
+        "\n",
+        "    Settings.embed_model = gemini_embedding_model\n",
+        "    Settings.llm = llm\n",
+        "\n",
+        "    return storage_context\n",
+        "\n",
+        "\n",
+        "def ingest_document():\n",
+        "    \"\"\"\n",
+        "    Using SimpleDirectoryReader, which creates documents out of every file in a given directory. It is built into LlamaIndex and can read a variety of formats including Markdown, PDFs, Word documents, PowerPoint decks, images, audio and video.\n",
+        "    \"\"\"\n",
+        "    documents = SimpleDirectoryReader(DOC_FOLDER).load_data()\n",
+        "    return documents"
```

Comment:
Does this need its own method? It's just one function.

Reply:
Removed. Added into the one liner.